tuf/api/metadata.py

```diff
@@ -524,21 +524,32 @@ class Role:
         unrecognized_fields: Dictionary of all unrecognized fields.
     """
 
+    @property
+    def keyids(self) -> str:
+        return self._keyids
+
+    # Preserve keyids order and enforce kids uniqueness.
+    @keyids.setter
+    def keyids(self, input_keyids: List[str]) -> None:
+        res_keyids = []
+        for keyid in input_keyids:
+            if keyid in res_keyids:
+                raise ValueError(
+                    f"keyids should be a list of unique strings,"
+                    f" instead got {input_keyids}"
+                )
```

Comment:
has to be a way to squeeze this into something less than four lines
```suggestion
                raise ValueError(f"Non-unique keyids: {input_keyids}"
```